Docs for ExternalJWTSigner (#48518)
kubernetes/website: content/en/docs/reference/access-authn-authz/service-accounts-admin.md

using the `--service-account-key-file` flag. The public key will be used to
verify the tokens during authentication.

{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}

An alternate setup to setting `--service-account-private-key-file` and `--service-account-key-file` flags is

> to configure a external-jwt-signer for [External ServiceAccount token signing and key management](#external-erviceAccount-token-signing-and-key-management). 

Reviewer: ```suggestion
to configure an external JWT signer for [external ServiceAccount token signing and key management](#external-erviceAccount-token-signing-and-key-management). 
```

Author: Done

> Note that both are mutually exclusive setups and cannot be configured together.

Reviewer: ```suggestion
Note that these setups are mutually exclusive and cannot be configured together.
```

Author: Done

```

## External ServiceAccount token signing and key management

{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}

The kube-apiserver can be configured to use external signer for token signing and token verifying key management.
This feature enables kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes) for service account credential signing and verification.
To configure kube-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a Unix domain socket (UDS) on a filesystem, or be prefixed with an @ symbol and name a UDS in the abstract socket namespace.
At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/release-1.32/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).
The external-jwt-signer must be healthy and be ready to serve supported service account keys for the kube-apiserver to start.

Check out [KEP-740](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/740-service-account-external-signing) for more details on ExternalJWTSigner.

> NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files unless --service-account-signing-endpoint is set; they are mutually exclusive ways of supporting JWT signing and authentication.

Reviewer: use the `note` shortcode?
consider wrap the flags with backticks?

kubectl -n examplens delete secret/example-automated-thing-token-zyxwv
```

## External ServiceAccount token signing and key management

{{< feature-state feature_gate_name="ExternalServiceAccountTokenSigner" >}}

> The kube-apiserver can be configured to use external-jwt-signer for token signing and key management.

Reviewer 2: ```suggestion
The kube-apiserver can be configured to use external signer for token signing and token verifying key management.
```

Author: Done

> This feature enables kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes).

Reviewer 2: ```suggestion
This feature enables kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes) for service account credential signing and verification.
```

Author: Done

> To configure kub-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a UDS on a filesystem, or

Reviewer 2: ```suggestion
To configure kube-apiserver to use external-jwt-signer set the `--service-account-signing-endpoint` flag to the location of a Unix domain socket (UDS) on a filesystem, or
```

Author: Done

be prefixed with an @ symbol and name a UDS in the abstract socket namespace.

> At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).

Reviewer 2: ```suggestion
At the configured UDS, shall be an RPC server which implements [ExternalJWTSigner](https://github.com/kubernetes/kubernetes/blob/release-1.32/staging/src/k8s.io/externaljwt/apis/v1alpha1/api.proto).
```

Author: Done

The external-jwt-signer must be healthy and be ready to serve supported service account keys for the kube-apiserver to start.

Check out [KEP-740](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/740-service-account-external-signing) for more details on ExternalJWTSigner.

> NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files (unless, --service-account-signing-endpoint is set; They are mutually exclusive ways of supporting JWT signing and authentication)

Reviewer 2: ```suggestion
NOTE: The kube-apiserver flags --service-account-key-file and --service-account-signing-key-file will continue to be used for reading from files unless --service-account-signing-endpoint is set; they are mutually exclusive ways of supporting JWT signing and authentication.
```

Author: Done
